Content/Items/Vitric/Weapons.CoachGunUpgrade.cs

```diff
@@ -0,0 +1,436 @@
+
+            Main.EntitySpriteDraw(texture,
+                Projectile.Center - Main.screenPosition + new Vector2(0f, Projectile.gfxOffY),
+                sourceRectangle, Color.White, Projectile.rotation, origin, Projectile.scale, spriteEffects, 0);
+
+            //drawing glow, well not anymore :(
+            /*Texture2D tex = ModContent.Request<Texture2D>(AssetDirectory.VitricItem + Name + "Glow").Value;
+
+            sourceRectangle = tex.Frame(1, Main.projFrames[Projectile.type], frameY: Projectile.frame);
+            origin = sourceRectangle.Size() / 2f;
+            origin.X = Projectile.spriteDirection == 1 ? sourceRectangle.Width - 21f : 21f;
+               
+            Main.spriteBatch.Draw(tex, Projectile.Center - Main.screenPosition, sourceRectangle, Color.White, Projectile.rotation, origin, 1, spriteEffects, 0);*/
+
+            return false;
+        }
+
+        public override bool OnTileCollide(Vector2 oldVelocity)
+        {
+            Helper.PlayPitched("GlassMiniboss/GlassBounce", 0.15f, Main.rand.NextFloat(-0.1f, 0.1f), Projectile.position);
+
+            if (Math.Abs(Projectile.velocity.X - oldVelocity.X) > float.Epsilon)
+                Projectile.velocity.X = -oldVelocity.X;
+
+            if (Math.Abs(Projectile.velocity.Y - oldVelocity.Y) > float.Epsilon)
+                Projectile.velocity.Y = -oldVelocity.Y;
+
+            return false;
+        }
+
+        public void ExplodeIntoShards(Vector2 velocity)
+        {
+            shrapnel = true;
+            if (Main.myPlayer == Projectile.owner)
+            {
+                for (int i = 0; i < 6; i++)
+                {
+                    Vector2 vel = (velocity.RotatedByRandom(MathHelper.ToRadians(8f))) * Main.rand.NextFloat(0.8f, 1.1f);
+                    Projectile.NewProjectile(Projectile.GetSource_Death(), Projectile.Center, vel * 1.45f, ModContent.ProjectileType<CoachGunUpgradeShards>(), (int)(Projectile.damage * 0.33f), 0.5f, Projectile.owner);
+                }
+                Projectile.NewProjectileDirect(Projectile.GetSource_FromThis(), Projectile.Center, Vector2.Zero, ModContent.ProjectileType<CoachGunUpgradeExplosion>(), (int)(Projectile.damage * 1.3f), 0f, Projectile.owner, 40, 1f);
+            }
+
+            for (int i = 0; i < 5; i++)
+            {
+                Vector2 vel = (velocity.RotatedByRandom(MathHelper.ToRadians(5f))) * Main.rand.NextFloat(0.2f, 0.4f);
+                Dust.NewDustDirect(Projectile.position, Projectile.width, Projectile.width, ModContent.DustType<VitricBombDust>(), vel.X, vel.Y, 0, default, Main.rand.NextFloat(0.7f, 0.9f)).noGravity = true;
+
+                for (int d = 0; d < 2; d++)
+                {
+                    Vector2 velo = (velocity.RotatedByRandom(MathHelper.ToRadians(12f))) * Main.rand.NextFloat(0.25f, 0.45f);
+                    Dust.NewDustPerfect(Projectile.Center, ModContent.DustType<Dusts.Glow>(),
```

Comment:
any reason this got kicked to the next line? this is far from the longest line around here